### PR TITLE
Add ContractDenyList

### PR DESCRIPTION
**Pull request by yjcbert**

### CODE REVIEW

File: modules/evm/eth.go [hunk 1]
```diff
@@ -640,3 +643,28 @@ func (esc EthSetupContextDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
 	newCtx = ctx.WithGasMeter(sdk.NewInfiniteGasMeter())
 	return next(newCtx, tx, simulate)
 }
+
+type EthCanCallDecorator struct {
```
Comment (zhangyelong) [inline modules/evm/eth.go:647]: `EvmContractCallableDecorator` seems better :)

File: app/ante.go [hunk 2]
```diff
@@ -75,7 +69,8 @@ func NewAnteHandler(options HandlerOptions) sdk.AnteHandler {
 				case "/ethermint.evm.v1.ExtensionOptionsEthereumTx":
 					// handle as *evmtypes.MsgEthereumTx
 					anteHandler = sdk.ChainAnteDecorators(
-						wevmante.NewEthCanCallDecorator(options.wevmKeeper),
+						perm.NewAuthDecorator(options.permKeeper),
+						appante.NewEthCanCallDecorator(options.permKeeper),
```
Comment (dreamer-zq) [inline app/ante.go:73]: It is recommended that the Decorator that is easy to check should be placed first, for the purpose of failing quickly

File: modules/opb/keeper/ante.go [hunk 3]
```diff
@@ -1,11 +1,10 @@
 package keeper
 
 import (
+	"github.com/bianjieai/irita/modules/opb/types"
```
Comment (dreamer-zq) [inline modules/opb/keeper/ante.go:4]: format code

File: modules/evm/eth.go [hunk 4]
```diff
@@ -640,3 +643,28 @@ func (esc EthSetupContextDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
 	newCtx = ctx.WithGasMeter(sdk.NewInfiniteGasMeter())
 	return next(newCtx, tx, simulate)
 }
+
+type EvmContractCallableDecorator struct {
+	keeper permkeeper.Keeper
```
Comment (dreamer-zq) [inline modules/evm/eth.go:648]: Tend to use interfaces to shield the dependencies of each module

**Review by aofengli — APPROVED** [review]

**Review by zhangyelong — APPROVED** [review]

**Review by aofengli — APPROVED** [review]

**Pull request merged**